pkg/coordinator/deposits.go

```diff
@@ -67,81 +59,179 @@ func ListDeposits(
 	return nil
 }
 
-func removeSwept(deposits []depositEntry) []depositEntry {
-	result := []depositEntry{}
-	for _, deposit := range deposits {
-		if deposit.isSwept {
-			continue
+// FindDepositsToSweep finds deposits that can be swept.
+// If a wallet public key hash is provided, it will find unswept deposits for the
+// given wallet. If a wallet public key hash is nil, it will check all wallets
+// starting from the oldest one to find a first wallet containing unswept deposits
+// and return those deposits.
+// maxNumberOfDeposits is used as a ceiling for the number of deposits in the
+// result. If number of discovered deposits meets the maxNumberOfDeposits the
+// function will stop fetching more deposits.
+// This function will return a wallet public key hash and a list of deposits from
+// the wallet that can be swept.
+// Deposits with insufficient number of funding transaction confirmations will
+// not be taken into consideration for sweeping.
+// The result will not mix deposits for different wallets.
+// TODO: Cache immutable data
```

Comment:
This would be a nice-to-have to optimize Ethereum and Bitcoin APIs usage. But I don't think we need to handle it in this PR.

Reply:
For example:
1. Cache the deposits that were already swept, so we don't call `GetDepositRequest` for them.
2. Cache the block number before which all deposits are swept so we can tighten the range for `PastDepositRevealedEvents` fetching.